detectable/src/test/java/com/synopsys/integration/detectable/detectables/clang/functional/DependencyListFileParserTest.java

```diff
@@ -0,0 +1,56 @@
+package com.synopsys.integration.detectable.detectables.clang.functional;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import com.synopsys.integration.detectable.detectables.clang.dependencyfile.DependenyListFileParser;
+
+public class DependencyListFileParserTest {
+
+    @Test
+    public void testSimple() {
+        final String curDirPath = System.getProperty("user.dir");
+        final File curDir = new File(curDirPath);
+        final File sourceFile = new File(curDir, "src/test/resources/detectables/functional/clang/src/process.c");
+        final File includeFile1 = new File(curDir, "src/test/resources/detectables/functional/clang/include/stdc-predef.h");
+        final File includeFile2 = new File(curDir, "src/test/resources/detectables/functional/clang/include/assert.h");
+        final String fileContents = String.format("dependencies: %s \\\n %s %s\\\n",
+            sourceFile.getAbsolutePath(), includeFile1.getAbsolutePath(), includeFile2.getAbsolutePath());
+
+        final DependenyListFileParser parser = new DependenyListFileParser();
+        final List<String> deps = parser.parseDepsMk(fileContents);
+
+        for (final String dep : deps) {
+            System.out.printf("dep: %s\n", dep);
+        }
+        assertTrue(deps.contains(sourceFile.getAbsolutePath()));
+        assertTrue(deps.contains(includeFile1.getAbsolutePath()));
+        assertTrue(deps.contains(includeFile2.getAbsolutePath()));
+    }
+
+    @Test
+    public void testNonCanonical() throws IOException {
+        final String curDirPath = System.getProperty("user.dir");
+        final File curDir = new File(curDirPath);
+        final File sourceFile = new File(curDir, "src/test/resources/detectables/functional/clang/src/process.c");
+        final File includeFile1 = new File(curDir, "src/test/resources/detectables/functional/clang/include/stdc-predef.h");
+        final File includeFile2 = new File(curDir, "src/test/resources/../../test/resources/detectables/functional/clang/include/assert.h");
```

Comment:
If these files are all empty below can we just create files in the unit test without including empty files in the resources?  The files created in the test would have paths similar to the paths after clang portion of the file paths currently.  

i.e. 
```
        final File testParent = new File(curDir,"test");
        final File canonicalTestParent = new File(testParent, "subDir1/subDir2);
        final File sourceFile = new File(testParent, "src/process.c");
        final File includeFile1 = new File(testParent, "include/stdc-predef.h");
        final File includeFile2 = new File(canonicalTestParent, "../../include/./assert.h");
        canonicalTestParent.mkdirs();
        sourceFile.create();
        includeFile1.create();
        includeFile2.create();
```
At the end of the test you can delete the files you created for this test.